### ISSUE TITLE
NetworkManager unable to save OpenVPN settings properly

### ISSUE THREAD

**Issue by otwieracz** [NixOS/nixpkgs#30235]
## Issue description

NetworkManager unable to save OpenVPN setting

### Steps to reproduce

Try to import previously working fine OpenVPN configuration - it will fail on cert file saving, setting `ca` key to both `ca` and `cert` and nothing to `private key`

## Technical details

* System: 18.03pre117327.3fe7cddc30 (Impala)
* Nix version: nix-env (Nix) 1.11.15
* Nixpkgs version: (run `"18.03pre117327.3fe7cddc30"`)
* Sandboxing enabled: yes

```
Oct 08 22:41:58 odyssey.otwiera.cz .nm-connection-[13784]: failed to commit changes to dconf: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name ca.desrt.dconf was not provided by any .service files
```

Possibly related to 
https://github.com/NixOS/nixpkgs/issues/2448

**Comment by otwieracz** [comment]
With `gnome.dconf` installed, it fails with:

```
Oct 08 22:49:06 odyssey.otwiera.cz gnome-keyring-daemon[32063]: couldn't access control socket: /run/user/1000/keyring/control: No such file or directory
Oct 08 22:49:06 odyssey.otwiera.cz .gnome-keyring-[32063]: couldn't access control socket: /run/user/1000/keyring/control: No such file or directory
```

**Comment by otwieracz** [comment]
After manual creation of `/etc/NetworkManager/system-connctions/my-vpn-file` it worked. However, still unable to save VPN config.

**Comment by BjornMelgaard** [comment]
try install `nix-env -i dconf`

**Comment by gebner** [comment]
I don't think this is dconf related, since nm-applet can create the vpn connection successfully.  It just set a wrong value for vpn.data:
```
$ nmcli conn edit vpn
nmcli> goto vpn
nmcli> goto data
nmcli> print
data: cert-pass-flags = 0, cert = /home/gebner/.vpn/ca.crt, ca = /home/gebner/.vpn/ca.crt, connection-type = tls, remote = ....
```

**Comment by gebner** [comment]
BTW, as a workaround you can fix the vpn connection manually using nmcli:
```
$ nmcli connection modify my-vpn vpn.data "connection-type = tls, key = /home/gebner/.vpn/some.key, ca = /home/gebner/.vpn/ca.crt, cert = /home/gebner/.vpn/some.crt, remote = example.com"
```

**Comment by baracoder** [comment]
(triage) Is this issue still present in 19.03?

**Comment by lloydhazlett** [comment]
@baracoder No, thankfully it is fixed for me in 19.03.
